services/configuration/src/tests/api.rs

```diff
@@ -286,6 +293,49 @@ fn test_votes_for_propose() {
     }
 }
 
+#[test]
+fn test_dissenting_votes_for_propose() {
+    let mut testkit: TestKit = TestKit::configuration_default();
+    let api = testkit.api();
+    // Apply a new configuration.
+    let cfg_change_height = Height(5);
+    let new_cfg = {
+        let mut cfg = testkit.configuration_change_proposal();
+        cfg.set_service_config("message", "First config change");
+        cfg.set_actual_from(cfg_change_height);
+        cfg.stored_configuration().clone()
+    };
+    let tx_propose = new_tx_config_propose(&testkit.network().validators()[0], new_cfg.clone());
+    let cfg_proposal_hash = new_cfg.hash();
+    assert_eq!(None, api.votes_for_propose(&new_cfg.hash()));
+    testkit.create_block_with_transactions(txvec![tx_propose]);
+    assert_eq!(
+        Some(vec![None; testkit.network().validators().len()]),
+        api.votes_for_propose(&new_cfg.hash())
+    );
+    // Push dissenting votes
+    let tx_dissenting_votes = testkit
+        .network()
+        .validators()
+        .iter()
+        .map(|validator| new_tx_config_vote_against(validator, cfg_proposal_hash))
+        .map(to_boxed)
+        .collect::<Vec<_>>();
+    testkit.create_block_with_transactions(tx_dissenting_votes);
+    let response = api.votes_for_propose(&new_cfg.hash())
+        .expect("Dissenting votes for config is absent");
+    for entry in response.into_iter().take(testkit.majority_count()) {
+        let tx = entry.expect("VoteAgainst for config is absent");
```

Comment:
I'd suggest to check that `entry` corresponds to the `VoteAgainst` variant.

Reply:
Done